webapp/channels/src/packages/mattermost-redux/src/reducers/entities/users.ts

```diff
@@ -500,6 +500,24 @@ function addToState<T>(state: Record<string, T>, key: string, value: T): Record<
     };
 }
 
+function dndEndTimes(state: RelationOneToOne<UserProfile, number> = {}, action: GenericAction) {
+    switch (action.type) {
+    case UserTypes.RECEIVED_STATUS: {
+        const userId = action.data.user_id;
+        const endTime = action.data.dnd_end_time;
+
+        return addToState(state, userId, endTime);
+    }
+    case UserTypes.RECEIVED_STATUSES: {
+        const userStatuses: UserStatus[] = action.data;
+
+        return userStatuses.reduce((nextState, userStatus) => addToState(nextState, userStatus.user_id, userStatus.dnd_end_time || 0), state);
+    }
```

Comment:
Could you also add the case for `PROFILE_NO_LONGER_VISIBLE` here as well?

I'm starting to wonder if we should just change `statuses` to store the whole status object or have a separate `currentUserStatus` since we only care about the `dnd_end_time` and the `manual` for the current user, but that's obviously outside the scope of this PR

Reply:
As of now, I have added case `PROFILE_NO_LONGER_VISIBLE`.

`statuses` change might affect multiple features